### DISCUSSION TITLE
OpenGL3+ Support on New Version

### DISCUSSION THREAD

**Discussion by justZZY** [openscenegraph/OpenSceneGraph#1201] [General]
I'm glad to hear that the OSG will restart maintain. These old versions are all based on fixed function pipeline. Some classes just like **osg::material** used old gl interface like **glMaterial** are ineffective when compile OSG with OpenGL3+. So is there a plan to fix the 
ineffective features with updated version? 

All the best.

**Comment by robertosfield** [comment]
The OSG is in maintenance mode not development mode.  It's about keeping existing features working rather significantly altering features or adding new major features.  

I have shelved the experimental work I did on providing a mapping from fixed function pipeline to shader work and plan to remove all if it from OpenSceneGraph master as I will not have the time to complete this, I also feel this is a dead end as modern scene graphs will be entirely shader based.  This work was never funded so I'm not about to resurrect a dead end bit of work on a legacy scene graph for no income. 


Presently all my income comes from VulkanSceneGarph so there is where the vast majority of my developer time will be spent.  The VulkanSceneGraph is built from the ground to handle modern approaches to rendering or modern graphics APIs unshackled by the legacy fixed function world of graphics.

The OSG is a now legacy scene graph built on a legacy graphics API (OpenGL) and is to be maintained in order to keep existing OSG based applications working.

If you want to modern things with the OSG then you can use OpenGL 3 + 4 features and ignore the fixed function pipeline, but you'll need to create your own uniform equivalents for things like materials and lights.   However, you might be better off just porting to the VSG as it does all this type of functionality far more cleanly and efficiently, partly because of Vulkan and partly because it's a modern scene graph not a legacy one.

**Reply by justZZY** [reply]
Ok, I know. Thanks for your reply. I believe vsg is the future too. So I'll try to learn your new framework and try to migrate my project.  👍 

Good Luck.
